database/misc/mongodb_user.py

```diff
@@ -261,14 +262,22 @@ def main():
         if password is None and update_password == 'always':
             module.fail_json(msg='password parameter required when adding a user unless update_password is set to on_create')
 
-        if update_password != 'always' and user_find(client, user):
+        uinfo = user_find(client, user, db_name)
+        if update_password != 'always' and uinfo:
             password = None
+            if list(map((lambda x: x['role']), uinfo['roles'])) == roles:
```

Comment:
imho, it will break the module with mongodb `2.4` version.
The `role` attribute is not present.
```
# use a fallback
list(map((lambda x: x['role']), uinfo.get('roles', [])))
```